src/query/query.rs

```diff
@@ -15,21 +16,24 @@ use crate::{
 
 const MAX_QUERY_LEN_TO_USE_GET: usize = 8192;
 
-pub use crate::cursors::{BytesCursor, RowCursor};
 use crate::headers::with_authentication;
 
+use crate::query::query_flags::QI;
+
 #[must_use]
 #[derive(Clone)]
-pub struct Query {
+pub struct Query<const INTERPFLAGS: u8 = { QI::DEFAULT }> {
```

Comment:
Changing a public type should be a major version bump.

Can this be avoided, by the way? What is the purpose of having `INTERPFLAGS` here instead of a regular setter?